test/SlowTests/Server/Documents/ETL/ConnectionStringTests.cs

```diff
@@ -276,5 +277,25 @@ public void CanGetConnectionStringByName()
                 Assert.True(resultElastic.ElasticSearchConnectionStrings.Count > 0);
             }
         }
+
+        [RequiresMsSqlFact]
+        public void CannotAddSqlConnectionStringWithInvalidFactoryName()
+        {
+            using (var store = GetDocumentStore())
+            {
+                Assert.Throws<RavenException>(() => store.Maintenance.Send(new PutConnectionStringOperation<SqlConnectionString>(new SqlConnectionString
+                {
+                    Name = "Invalid Factory Connection String",
+                    ConnectionString = "some-connection-string-that-doesnt-matter",
+                    FactoryName = "Invalid.Factory.4.20-final.stable"
+                })));
+                Assert.Throws<RavenException>(() => store.Maintenance.Send(new PutConnectionStringOperation<SqlConnectionString>(new SqlConnectionString
```

Comment:
Please check the exception messages. Just checking that we get any exception is not sufficient

Reply:
fixed